client/src/components/RecipeShow.js

```diff
@@ -11,26 +16,140 @@ const RecipeShow = (props) => {
       const response = await fetch(`/api/v1/users/recipes/${id}`)
       const body = await response.json()
       setRecipe(body.recipe)
+      setRecipeIngredients(body.recipe.ingredients)
     } catch(err) {
       console.error(`Error in Fetch: ${err.message}`)
     }
   }
 
+  const fetchPantryIngredients = async () => {
+    try {
+      const response = await fetch(`/api/v1/users/ingredients/all`)
+      const body = await response.json()
+      setPantryIngredients(body.ingredients)
+    } catch(err) {
+      console.error(`Error in fetch: ${err.message}`)
+    }
+  }
+
   useEffect(() => {
       fetchRecipeDetails()
-  }, [])
+      fetchPantryIngredients()
+    }, [])
 
-  let ingredientsList
-  if(recipe.ingredients) {
-    ingredientsList = recipe.ingredients.map(ingredient => {
+    useEffect(() => {
+      if(recipeIngredients.length > 0) {
+        getNewTotals()
+      }
+    }, [recipeIngredients, pantryIngredients])
+
+    const getNewTotals = () => {
+      let ingredientsToUpdate = recipeIngredients.map(ingredient => {
+        const ingredientInPantry = pantryIngredients.find(pantryIngredient => pantryIngredient.name.toLowerCase() === ingredient.name)
+        if(ingredientInPantry) {
+          if(ingredientInPantry.unit === ingredient.unit) {
+            const updatedIngredient = {
+              ...ingredientInPantry,
+              amount: ingredientInPantry.amount - ingredient.amount
+            }
+            return updatedIngredient
+            } else {
+            const convertedRecipeIngredient = Ingredients.convertIngredient(ingredient)
+            const convertedPantryIngredient = Ingredients.convertIngredient(ingredientInPantry)
+            if (!convertedRecipeIngredient || !convertedPantryIngredient) {
+              return false
+            } else {
+              const newTotal = Ingredients.convertAndUpdate(ingredientInPantry, ingredient)
+              const updatedIngredient = {
+                ...ingredientInPantry,
+                amount: newTotal
+              }
+              return updatedIngredient
+            }
+          }
+        } else {
+          return false
+        }
+      })
+      setUpdatedIngredients(ingredientsToUpdate)
+    }
```

Comment:
You've done a great job adding an Ingredient POJO...but you could actually store all of this logic in ANOTHER POJO that manages this logic

IngredientConver.js & IngredientReconciler.js

However, you could also house all of this logic, including your POJO that you made....on the backend. And then call on these POJOs in your serializer. Then your frontend receives data for:
- recipe and ingredients for this recipe
- ingredients for the user's pantry
- data on whether or not the user can make the recipe
...All retrieved in ONE fetch request